CI failure in io module

https://travis-ci.org/github/pygae/clifford/jobs/751737929#L369-L388

This looks like a possible bug in h5py:

https://github.com/pygae/clifford/blob/a1b1ba3519612af4d0e0c842ff290ee0f5d25ca1/clifford/test/test_io.py#L22-L35
```
>       np.testing.assert_equal(basis_names, basis_names_2)
E       AssertionError: 
E       Arrays are not equal
E       
E       Mismatched elements: 5 / 5 (100%)
E        x: array(['e1', 'e2', 'e3', 'e4', 'e5'], dtype='<U2')
E        y: array([b'e1', b'e2', b'e3', b'e4', b'e5'], dtype=object)
```
Somehow the array is being converted from `str` to `bytes`...

Versions:

* numpy: 1.19.4
* h5py: 3.1.0
* python: 3.8.6

--- a/clifford/io.py
+++ b/clifford/io.py
@@ -136,5 +136,9 @@ def read_ga_file(file_name):
         else:
             support = None
         metric = f['metric'][:]
-        basis_names = f['basis_names'][:]
+        if hasattr(h5py.Dataset, 'asstr'):
+            basis_names = f['basis_names'].asstr()[:]
+        else:
+            # h5py < 3
+            basis_names = f['basis_names'][:]
     return data_array, metric, basis_names, support

